UVHub/src/main/java/com/envirover/uvhub/StateCodec.java

```diff
@@ -51,108 +48,175 @@
         msg_global_position_int msg = new msg_global_position_int();
         msg.sysid = state.sysid;
-        msg.alt = state.altitude_amsl * 1000;
+        msg.time_boot_ms = state.timestamp;
+        msg.alt = state.altitude * 1000;
         msg.lat = state.latitude;
         msg.lon = state.longitude;
         msg.hdg = state.heading;
-        msg.relative_alt = state.altitude_sp * 1000;
+        msg.relative_alt = state.target_altitude * 1000;
         return msg;
     }
 
-    private static MAVLinkMessage getMissionCurrentMsg(msg_high_latency state) {
+    private static MAVLinkMessage getMissionCurrentMsg(msg_high_latency2 state) {
         msg_mission_current msg = new msg_mission_current();
         msg.sysid = state.sysid;
         msg.seq = state.wp_num;
         return msg;
     }
 
-    private static MAVLinkMessage getNavControllerOutputMsg(msg_high_latency state) {
+    private static MAVLinkMessage getNavControllerOutputMsg(msg_high_latency2 state) {
         msg_nav_controller_output msg = new msg_nav_controller_output();
         msg.sysid = state.sysid;
-        msg.nav_bearing = (short) (state.heading_sp / 100);
+        msg.nav_bearing = (short) (state.target_heading * 2);
         return msg;
     }
 
-    private static MAVLinkMessage getVfrHudMsg(msg_high_latency state) {
+    private static MAVLinkMessage getVfrHudMsg(msg_high_latency2 state) {
         msg_vfr_hud msg = new msg_vfr_hud();
         msg.sysid = state.sysid;
-        msg.airspeed = state.airspeed;
-        msg.alt = state.altitude_amsl;
-        msg.climb = state.climb_rate;
-        msg.groundspeed = state.groundspeed;
-        msg.heading = (short) (state.heading / 100);
+        msg.airspeed = state.airspeed / 5;
+        msg.alt = state.altitude;
+        msg.climb = state.climb_rate / 10;
+        msg.groundspeed = state.groundspeed / 5;
+        msg.heading = (short)(state.heading * 2);
         msg.throttle = state.throttle;
         return msg;
     }
 
-    private static MAVLinkMessage getBattery2Msg(msg_high_latency state) {
+    private static MAVLinkMessage getBattery2Msg(msg_high_latency2 state) {
         msg_battery2 msg = new msg_battery2();
         msg.sysid = state.sysid;
         msg.current_battery = -1;
-        msg.voltage = state.temperature_air * 1000;
+        msg.voltage = state.custom1 * 1000;
         return msg;
     }
 
+    private static MAVLinkMessage getWindMsg(msg_high_latency2 state) {
+        msg_wind msg = new msg_wind();
+        msg.sysid = state.sysid;
+        msg.direction = state.wind_heading * 2;
+        msg.speed = (float)(state.windspeed / 5.0);
+        return msg;
+    }
+
+    private static MAVLinkMessage getScaledPressure(msg_high_latency2 state) {
+        msg_scaled_pressure msg = new msg_scaled_pressure();
+        msg.sysid = state.sysid;
+        msg.temperature = state.temperature_air;
+        return msg;
+    }
+
+    private static long getHealth(int failure_flags) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 34. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.